### PR TITLE
Block new PopAds variant by regex

### PR DESCRIPTION
**Pull request by Yuki2718**
### To avoid invalid pull requests, please check and confirm following terms

- [x] This is not an ad/bug report;
- [x] My code follows the [guidelines](https://github.com/AdguardTeam/AdguardFilters/blob/master/CONTRIBUTING.md) and [syntax](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) of this project;
- [x] I have performed a self-review of my own changes;
- [x] My changes do not break web sites, apps and files structure.

### If the problem does not fall under any category that is listed here, please write a comment below in corresponding section

- [x] Missed ads or ad leftovers;
- [ ] Website or app doesn't work properly;
- [ ] AdGuard gets detected on a website;
- [ ] Missed analytics or tracker;
- [ ] Social media buttons — share, like, tweet, etc;
- [ ] Annoyances — pop-ups, cookie warnings, etc;
- [ ] Filters maintenance.

### Enter the issue address

ex. https://github.com/AdguardTeam/AdguardFilters/issues/167344

### Add your comment and screenshots

They are now serious about bypassing blocker. EL is tbh behind their pace of revolving domain, and even scriptlet/HTML filtering may not last long as they constantly change code structure, not to mention race condition and iOS without Advanced protection.

https://regex101.com/r/ZddUVH/1

Known false positive by the rule: https://github.com/uBlockOrigin/uAssets/issues/19848 but it should be okay if limited to specific domains. I can't test the regex for iOS.

### Terms

- [x] By submitting this issue, I agree that pull request does not contain private info and all conditions are met

### CODE REVIEW

**Comment by Yuki2718** [comment]
These rules are also needed:
```
://www.*.css|$script,third-party
://www.*/images/*.min.js|$script,third-party
://www.*.com/css/$script,third-party,domain=~principal.com|~principalcdn.com
://www.*.com/js/css/$script,third-party
```
but you may want to add `domain=` for them too given [false positive](https://github.com/uBlockOrigin/uAssets/issues/20983).

File: BaseFilter/sections/specific.txt [hunk 1]
```diff
@@ -125,6 +125,10 @@ kimcartoon.me##div > div[id][style*="overflow: hidden"][style*="width: 300px;"][
...
+!+ PLATFORM(ios, ext_safari)
+/^https:\/\/www\.[a-z][a-z][a-z][a-z][a-z][a-z][a-z][a-z]+\.com\/(static\/)?(common\/)?(assets\/)?(scripts\/)?(api\/)?(js\/)?([0-9A-Za-z]+[-.])*[0-9A-Za-z]+\.min\.js$/$script,third-party,match-case,domain=560pmovie.com|afrohung.com|bet36.es|blackcockadventure.com|blackcockchurch.org|blog.fc2.com|blogspot.com|blogtrabalhista.com|camgirlcum.com|castingx.net|compupaste.com|derleta.com|desixx.net|diskusscan.com|dixva.com|dlhd.sx|ekasiwap.com|freetvsports.com|gameronix.com|gamesmountain.com|girlscanner.org|giurgiuveanul.ro|hayhd.net|hopepaste.download|igfap.com|incestflix.com|jp-films.com|kaoskrew.org|mdy48tn97.com|multicanaistv.com|novelas4k.com|nudebabesin3d.com|offlink.us|pervyvideos.com|phim12h.com|picsxxxporn.com|pinayscandalz.com|playtube.co.za|pornfits.com|pornhoarder.net|pornodominicano.net|pover.org|pubfilmz.com|rapbeh.net|rapload.org|rapelust.com|robaldowns.com|scat.gold|scatfap.com|scatkings.com|seneporno.com|shadowrangers.net|shahee4u.cam|sportea.online|stream.lc|streamcenter.pro|streamhub.to|subtitleporn.com|taboosex.club|vidtapes.com|watchjavidol.com|wincest.xyz|xxxxvideo.uno
```
Comment (Alex-302) [inline BaseFilter/sections/specific.txt:131]: Safari rule requires also gameshdlive.net (main frame). 
But the rule is working.

**Review by Alex-302 — CHANGES_REQUESTED** [review]

**Comment by Alex-302** [comment]
> These rules are also needed:


But they are too wide.

**Comment by Yuki2718** [comment]
> But they are too wide.

Okay, then let's scope with domain=. In that case, it will be more convenient to integrate `(?:images\/)?` to the current regex and merge the rest three rules to single regex given all of them have `css` so that the resultant rules are just two where we can only add domain (+ occarional maintanance). I will craft the regex today by looking up more examples.

**Comment by Yuki2718** [comment]
Samples:
```
https://www.omgrdrodobidu.com/images/static/ounng-ckeditor.min.js
https://www.jqcyacoxrvada.com/css/nocURI.min.js
https://www.pcdgninekvch.com/api/js/hwtinstafetch.min.css
https://www.kewnemhpbmzkm.com/static/common/xajchess.min.css
​https://www.pbqqzibusu.com/assets/uemd3.chart.min.js
https://www.pbqqzibusu.com/common/assets/gsiui.min.js
https://www.kewnemhpbmzkm.com/css/images/kxmradian.min.css
https://www.omgrdrodobidu.com/scripts/api/fcyTreant.min.js
https://www.pcdgninekvch.com/gambackbone-min.js
https://www.ifmjzbdyk.com/js/css/xsosurvey.angular.min.js
https://www.wdevxtmasfdswx.com/assets/scripts/wmnbootstrap-colorpicker-module.min.css
https://www.fyvdxqufaxkli.com/gytleaflet.toolbar.css
https://www.pbqqzibusu.com/api/hdgzabuto_calendar.min.js
```
I had to iterate `(?:js\/)?` to cover all patterns. Unfortunately, there are some cases not ending with `min.js`:
```
https://www.pcdgninekvch.com/images/miuviz.js
https://www.ifmjzbdyk.com/hjpleaflet.toolbar.js
https://www.jqcyacoxrvada.com/css/ycfeditor.js
https://www.txhrnluuyt.com/static/common/dotdotdot.js
```
but specifying `min` greatly reduces false positive, therefore I just added these patterns.

**Comment by Alex-302** [comment]
> Okay, then let's scope with domain=

Ok.

Please add a link of PR as a comment, and a short explanation how/when to change them (not everyone is aware of these rules)

**Comment by Alex-302** [comment]
@AdamWr What do you think?

**Comment by AdamWr** [comment]
Looks good to me.

Additional note, not needed to read, just my thoughts:
<details>

Maybe this part:
```
(?:js\/)?(?:css\/)?(?:images\/)?(?:static\/)?(?:common\/)?(?:assets\/)?(?:scripts\/)?(?:api\/)?(?:js\/)?
```
could be changed to:
```
(?:(?:js|css|images|static|common|assets|scripts|api)\/)*
```
rule:
```
/^https:\/\/www\.[a-z]{8,16}\.com\/(?:(?:js|css|images|static|common|assets|scripts|api)\/)*(?:[0-9A-Za-z_]{1,20}[-.])*[0-9A-Za-z_]{1,20}\.(?:min|miuviz|dotdotdot|toolbar|[a-z]{3}editor)\.js$/$domain=[...]
```
Basing on provided examples, it covers the same cases and I think that it would be easier to add new value to it, like `...|api|newValue` than adding `(?:newValue\/)?`.

But it's not necessary and I'm not sure if it will match exactly the same things like original rule.

In the case if regexp doesn't match something, we can use rule with `denyallow`, I guess it should works fine in some cases.

</details>

**Comment by Yuki2718** [comment]
@AdamWr Yeah, I thought about that and may eventually change to that, but so far there seem to be some fixed patterns of the order they align. I thought `(?:(?:js|css|images|static|common|assets|scripts|api)\/)*` might be too generic as it can match whatever orders and numbers of them, probably my overthinking though.

**Review by Alex-302 — APPROVED** [review]

**Comment by Yuki2718** [comment]
> Please add a link of PR as a comment, and a short explanation how/when to change them (not everyone is aware of these rules)

Forgot this, added comment

**Comment by Yuki2718** [comment]
I removed the old rules as I no longer see them, should be replaced by new rules in https://github.com/AdguardTeam/AdguardFilters/pull/175819. `://www.*.com/*.css|$script,third-party,domain=` can too be removed once v4.4 came out.

**Pull request merged**